compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -350,11 +368,24 @@ LogicalResult AIETargetBackend::serializeExecutable(
     {
       SmallVector<StringRef> cmdEnvRefs{cmdEnv.begin(), cmdEnv.end()};
       int result = llvm::sys::ExecuteAndWait(cmdArgs[0], cmdArgs, cmdEnvRefs);
-      if (result != 0)
+      if (result != 0 && AttemptingMerge) {
```

Comment:
Can we detect before hand if the same kernel appears twice? Just because it'll be better to know why we failed. My concern is that we'll stop being able to merge for some other reason, and not know about it.

Reply:
I am not sure I correctly understand the case you are thinking about. But if I understand correctly you are thinking some kernel cannot be merged in the existing XCLBIN and if the next kernel is same as the one that couldnt be we will just not be able to merge?

If what I interpret is correct then a few reasons why this is not of concern.
1. IREE has deduplication passes after dispatch formation so same kernel is never appearing twice.
2. The success of the merge is independent of the content in the kernel. The only case that I am aware of when it will fail is when we exceed the number of PDIs (we have one PDI per kernel for now until we get to super kernels) that xclbin support. I believe the limit is 16. So my idea here is the 17th kernel will fail and then start a new xclbin, and the process repeats for the 33rd kernel and so on. A bit hard to test and we arent at that scale anyway now so havent worked on testing that.